src/charactersheet/viewmodels/common/party_manager/index.js

```diff
@@ -1,282 +1,104 @@
-/*eslint no-console:0*/
-import {
-    AuthenticationToken,
-    ChatRoom
-} from 'charactersheet/models/common';
-import {
-    ChatServiceManager,
-    PersistenceService,
-    XMPPService
-} from 'charactersheet/services/common';
-import {
-    CoreManager,
-    Fixtures,
-    Notifications
-} from 'charactersheet/utilities';
-import ko from 'knockout';
+import autoBind from 'auto-bind';
+import { observable, components, pureComputed } from 'knockout';
+import { CoreManager, Notifications } from 'charactersheet/utilities';
+import { PartyService } from 'charactersheet/services';
+import { ViewModel } from 'charactersheet/viewmodels/abstract';
 import template from './index.html';
 
-export function PartyManagerViewModel() {
-    var self = this;
 
-    self.loggedIn = ko.observable(false);
-    self.inAParty = ko.observable(false);
-    self.parties = ko.observableArray([]);
+export class PartyManagerViewModel extends ViewModel {
 
-    self.createOrJoin = ko.observable('create');
+    constructor() {
+        super();
+        autoBind(this);
+        this.party = observable(PartyService.party);
+        this.shortCode = observable();
```

Comment:
Why did we call this short code? isnt this just the full name of the party?

Reply:
Technically the party doesn't have a name or the name is UUID. This is more of a "Join Code".

Reply:
isn't think the party name created in the api? the `word-word-####` or whatever?